### PR TITLE
Fix absolute icon name

### PR DESCRIPTION
**Pull request by johanmalm**
@vyivel @tokyo4j would one of you mind doing a review of this one.
I've tested and it works with `gtk-lshw` which has an absolute icon-name.

### CODE REVIEW

File: src/icon-loader.c [hunk 1]
```diff
@@ -101,37 +143,35 @@ icon_loader_lookup(struct server *server, const char *app_id, int size, int scal
 		icon_name = app_id;
```
Comment (vyivel) [inline src/icon-loader.c:156]: This creates an interesting case where an application can set its app_id to an absolute path and get it loaded as its icon. An absolute path must only come from `sfdo_desktop_entry_get_icon()`.

File: src/icon-loader.c [hunk 2]
```diff
@@ -120,7 +120,20 @@ process_rel_name(struct icon_ctx *ctx, const char *icon_name,
 static int
 process_abs_name(struct icon_ctx *ctx, const char *icon_name)
 {
-	wlr_log(WLR_ERROR, "absolute paths not yet supported");
+	ctx->path = xstrdup(icon_name);
+	if (str_endswith(icon_name, "png")) {
```
Comment (vyivel) [inline src/icon-loader.c:124]: I feel like checking for ".png" (with a dot) would be slightly more robust.

File: src/icon-loader.c [hunk 3]
```diff
@@ -82,6 +83,60 @@ icon_loader_finish(struct server *server)
...
+#endif
+	struct sfdo_icon_file *icon_file = sfdo_icon_theme_lookup(
+		loader->icon_theme, icon_name, SFDO_NT, size, scale,
+		lookup_options);
+	if (!icon_file || icon_file == SFDO_ICON_FILE_INVALID) {
+		ret = -1;
+		goto out;
+	}
+	const char *path = sfdo_icon_file_get_path(icon_file, NULL);
+	if (string_null_or_empty(path)) {
```
Comment (vyivel) [inline src/icon-loader.c:109]: This should always be `false`.

File: src/icon-loader.c (hunk 1 above)
Comment (johanmalm) [reply src/icon-loader.c]: Yes, that's a weird one. Have tweaked logic to avoid that.

File: src/icon-loader.c (hunk 2 above)
Comment (johanmalm) [reply src/icon-loader.c]: Good point. Done.

File: src/icon-loader.c (hunk 3 above)
Comment (johanmalm) [reply src/icon-loader.c]: Okay, have just gone for a straight `xstrdup(sfdo_icon_file_get_path())`

I did actually look at the `sfdo_icon_file_get_path()` source - but I'm just paranoid :smile: I'll try to relax.

**Comment by vyivel** [comment]
LGTM

**Comment by johanmalm** [comment]
Thanks. Will rebase and go.

**Comment by 01micko** [comment]
Works well.

**Pull request merged**